### PR TITLE
add a script to automate the release process

### PR DESCRIPTION
**Pull request by roycaihw**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
Add a script that automate the release process. It should be used when we want to:
- in the master branch, create a new snapshot for a new Kubernetes minor release
- in the master branch, update an existing snapshot for a new Kubernetes patch release
   - in both cases above, update python-base if needed, and collect release notes for python orr python-base changes. 
- in a release branch, pull the master branch a cut a release with only version constant changes; send a separate PR to reflect the new release in the master branch CHANGELOG / README

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: scripts/apply-hotfixes.sh [hunk 1]
```diff
@@ -24,16 +24,6 @@ then
     exit 1
 fi
 
-# Check if the current branch is a release branch (release-*)
-# If it is not a release branch, don't let the patch be applied
```
Comment (roycaihw) [inline scripts/apply-hotfixes.sh:28]: we need to apply hotfixes when generating snapshots in the master branch

File: scripts/release.sh [hunk 2]
```diff
@@ -1,10 +1,126 @@
...
+# - add user input validation
+# - add function input validaiton (release/version strings start with 'v' or not)
+# - automatically send a PR; provide useful links for review
+#   - master branch diff: https://github.com/kubernetes-client/python/compare/commit1..commit2
+#   - python base diff: https://github.com/kubernetes-client/python-base/compare/commit1..commit2
+#   - Kubernetes changelog, e.g. https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.18.md
+# - add debug log
+# - add a sentence about "changes since {last release}". In most cases our
+#   releases should be sequential. This script (the workflow above) is based on
+#   this assumption, and we should make the release note clear about that.
```
Comment (roycaihw) [inline scripts/release.sh:58]: @yliaog This (and the Workflow above) is the sequential releases I mentioned in https://github.com/kubernetes-client/python/pull/1425#discussion_r618788294.

Comment (yliaog) [reply scripts/release.sh:58]: ok. let me know once the PR is ready for review.

Comment (roycaihw) [reply scripts/release.sh:58]: I generated a 19.0.0-snapshot client using this script: https://github.com/kubernetes-client/python/pull/1547

I think we can submit the current progress and develop on top of it. Please take a look.

File: scripts/release.sh [hunk 3]
```diff
@@ -1,10 +1,128 @@
...
+#     python / python-base change in the release note
+#   - API change w/ release notes
+#   - master change w/ release notes
+#   - submodule change w/ release notes
+# 3. [release branch] create a new release
+#   - pull master
+#     - it's possible that master has new changes after the latest snaphost,
+#       update CHANGELOG accordingly
+#   - for generated file, resolve conflict by committing the master version
+#   - abort if a snapshot doesn't exists
```
Comment (yliaog) [inline scripts/release.sh:38]: s/exists/exist/

File: scripts/release.sh [hunk 4]
```diff
@@ -1,10 +1,128 @@
...
+# get API change release notes since $old_k8s_api_version.
+# NOTE: $old_k8s_api_version may be one-minor-version behind $KUBERNETES_BRANCH, e.g.
+#   KUBERNETES_BRANCH=release-1.19
+#   old_k8s_api_version=1.18.17
+# when we bump the minor version for the snapshot in the master branch. We
+# don't need to collect release notes in release-1.18, because any API
+# change in 1.18.x (x > 17) must be a cherrypick that is already included in
+# release-1.19.
+# TODO(roycaihw): not all Kubernetes API changes modify the OpenAPI spec.
+# Download the patch and skip if the spec is not modified. Also we want want to
```
Comment (yliaog) [inline scripts/release.sh:101]: s/want//

File: scripts/release.sh (hunk 3 above)
Comment (roycaihw) [reply scripts/release.sh]: Done

File: scripts/release.sh (hunk 4 above)
Comment (roycaihw) [reply scripts/release.sh]: Done

**Pull request merged**